### ISSUE TITLE
Replace `execution_date` in `TI.generate_command` to send run_id instead

### ISSUE THREAD

**Issue by kaxil** [apache/airflow#16303]
Currently we use execution-date when generating the command to send via executor in https://github.com/apache/airflow/blob/9c94b72d440b18a9e42123d20d48b951712038f9/airflow/models/taskinstance.py#L420-L436

We should replace the execution_date in their to run_id instead and handle the corresponding changes needs on the executor and worker.

**Comment by uranusjr** [comment]
Depends on #16030.

**Comment by ashb** [comment]
@SamWheating You said you'd like to help on AIP-39? This might be an easy one to get started with (though we can't merge a PR for this change until #16030 is merged)

**Comment by SamWheating** [comment]
For sure, feel free to assign this one to me and I can look at it this week.

**Comment by ashb** [comment]
@SamWheating See also https://github.com/apache/airflow/pull/7085 -- you might want to fold that in to this change.

**Comment by SamWheating** [comment]
I think that this depends on https://github.com/apache/airflow/issues/16303 as well, since otherwise the `run_id` isn't directly available within a TaskInstance (and doing something like `TaskInstance.get_dagrun().run_id` doesn't feel like a proper solution)

How should I proceed?
- Fold that issue into this one (in a single PR)
- Wait for that one to complete first
- Take on that issue in a separate PR (as its currently unassigned)

**Comment by uranusjr** [comment]
Since #16302 will likely be pretty large, I would try to avoid depending on it if possible. One possibility is to create a `@property` (or SQLAlchemy relationship since we may to filter on it? not sure) on `TaskInstance` that uses `execution_date` and `dag_id` underneath, and add a `TODO` comment for #16302 to change the implementation when it implements the actual `run_id` field. Not 100% sure if it’s doable, but may be worth a try.

**Comment by Dr-Denzy** [comment]
Another test.
